Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -274,7 +272,29 @@ private static bool IsIonNamedProperty(PropertyInfo property)
         {
             return property.GetCustomAttribute(typeof(IonPropertyName)) != null;
         }
-        
+
+        /// <summary>
+        /// We only serialize public, internal, and protected internal gettable properties or properties with IonPropertyName annotation.
+        /// </summary>
+        private static bool HasValidAccessModifierForGetter(PropertyInfo propertyInfo)
+        {
+            var methodInfo = propertyInfo.GetGetMethod(true);
+
+            return methodInfo != null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly
+                || propertyInfo.GetCustomAttribute(typeof(IonPropertyName)) != null);
+        }
+
+        /// <summary>
+        /// We only deserialize public, internal, and protected internal settable properties or properties with IonPropertyName annotation.
+        /// </summary>
+        private static bool HasValidAccessModifierForSetter(PropertyInfo propertyInfo)
+        {
+            var methodInfo = propertyInfo.GetSetMethod(true);
+
+            return methodInfo != null && (methodInfo.IsPublic || methodInfo.IsAssembly || methodInfo.IsFamilyOrAssembly
+                || propertyInfo.GetCustomAttribute(typeof(IonPropertyName)) != null);
+        }
```

Comment:
Can we reduce the duplication between these two methods? (although not a big deal given how short these methods are.)

Reply:
Good idea! Reduced to one function.